pre_commit/languages/golang.py

```diff
@@ -145,6 +145,7 @@ def install_environment(
     env = no_git_env(dict(os.environ, GOPATH=gopath))
     env.pop('GOBIN', None)
     if version != 'system':
+        env['GOTOOLCHAIN'] = 'local'
```

Comment:
I think (?) this should be part of `get_env_patch` ?

Reply:
From what I can see `get_env_patch` is only called from `in_env` which is used when running the hook and not during the installation. This should only be relevant during installation because that's when the build happens. 

I ran the test to make sure it works this way and it breaks when I move it.

Reply:
hmmm I guess? but what about for hooks that just call `gofmt` -- I know the rust toolchain gets grumbly when the versions mismatch -- does something like this survive?

```yaml
-   repo: local
    hooks:
    -   id: gofmt
        name: gofmt
        language: golang
        entry: gofmt -l -w
        types: [go]
```

Reply:
`gofmt` doesn't seem to care about this option, but `go fmt` performs the toolchain switch. Looking at the code it seems that any `go` command will try to perform a [toolchain switch](https://github.com/golang/go/blob/bae2e968e2daadd39c1bdb1221648361d7277ddc/src/cmd/go/main.go#L105).

I guess the variable should be set in both places then.